core/src/main/kotlin/com/uber/sdk2/core/ui/UberButton.kt

```diff
@@ -15,243 +15,53 @@
-  private fun setPaddingAttributes(
-    context: Context,
-    attrs: AttributeSet?,
-    defStyleAttr: Int,
-    defStyleRes: Int,
-  ) {
-    val attrsResources =
-      intArrayOf(
-        android.R.attr.padding,
-        android.R.attr.paddingLeft,
-        android.R.attr.paddingTop,
-        android.R.attr.paddingRight,
-        android.R.attr.paddingBottom,
-      )
-    val paddingAttributes =
-      context.theme.obtainStyledAttributes(attrs, attrsResources, defStyleAttr, defStyleRes)
-    try {
-      val padding = paddingAttributes.getDimensionPixelOffset(0, 0)
-      var paddingLeft = paddingAttributes.getDimensionPixelSize(1, 0)
-      paddingLeft = if (paddingLeft == 0) padding else paddingLeft
-      var paddingTop = paddingAttributes.getDimensionPixelSize(2, 0)
-      paddingTop = if (paddingTop == 0) padding else paddingTop
-      var paddingRight = paddingAttributes.getDimensionPixelSize(3, 0)
-      paddingRight = if (paddingRight == 0) padding else paddingRight
-      var paddingBottom = paddingAttributes.getDimensionPixelSize(4, 0)
-      paddingBottom = if (paddingBottom == 0) padding else paddingBottom
-      setPadding(paddingLeft, paddingTop, paddingRight, paddingBottom)
-    } finally {
-      paddingAttributes.recycle()
+/** A button that looks like the Uber button. */
+@Composable
+fun UberButton(text: String, isWhite: Boolean = false, onClick: () -> Unit) {
+  val interactionSource = remember { MutableInteractionSource() }
+  val isPressed = interactionSource.collectIsPressedAsState().value
+  val backgroundColor =
+    if (isPressed) {
+      if (isWhite) colorResource(id = R.color.uber_white_40)
+      else colorResource(id = R.color.uber_black_90)
+    } else {
+      if (isWhite) colorResource(id = R.color.uber_white)
+      else colorResource(id = R.color.uber_black)
     }
-  }
 
-  @SuppressLint("ResourceType")
-  private fun setTextAttributes(
-    context: Context,
-    attrs: AttributeSet?,
-    defStyleAttr: Int,
-    defStyleRes: Int,
-  ) {
-    val attrsResources =
-      intArrayOf(
-        android.R.attr.textColor,
-        android.R.attr.gravity,
-        android.R.attr.textStyle,
-        android.R.attr.text,
-      )
-    val textAttributes =
-      context.theme.obtainStyledAttributes(attrs, attrsResources, defStyleAttr, defStyleRes)
-    try {
-      setTextColor(textAttributes.getColor(0, Color.WHITE))
-      gravity = textAttributes.getInt(1, Gravity.CENTER)
-      typeface = Typeface.defaultFromStyle(textAttributes.getInt(2, Typeface.NORMAL))
-      val text = textAttributes.getString(3)
-      if (text != null) {
-        setText(text)
-      }
-    } finally {
-      textAttributes.recycle()
+  val textColor =
+    if (isWhite) {
+      colorResource(id = R.color.uber_black)
+    } else {
+      colorResource(id = R.color.uber_white)
     }
-  }
 
-  companion object {
-    @StyleRes private val STYLES = intArrayOf(R.style.UberButton, R.style.UberButton_White)
-  }
+  Text(
```

Comment:
why not use the Button composable? This has ColorDefaults that can be overridden.